include/pybind11/cast.h

```diff
@@ -1610,11 +1610,11 @@ PYBIND11_NAMESPACE_BEGIN(detail)
 struct function_record;
 
 // forward declaration (definition in pybind11.h)
-std::string generate_function_signature(const char *type_caster_name_field,
-                                        function_record *func_rec,
-                                        const std::type_info *const *types,
-                                        size_t &type_index,
-                                        size_t &arg_index);
+inline std::string generate_function_signature(const char *type_caster_name_field,
```

Comment:
Could you please remove the `inline` here again?

A few years ago I asked around and got strong advice from a bunch of experts: Don't use `inline` in function declarations, only definitions.